lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3509,45 +3480,44 @@ protected static function get_block_element_selectors( $root_selector ) {
 		return $element_selectors;
 	}
 
-
 	/**
-	 * Generates style declarations for the block's features e.g. color, border,
-	 * typography etc, that have custom selectors in their block metadata.
+	 * Generates style declarations for a node's features e.g. color, border,
+	 * typography etc, that have custom selectors in their related block's
+	 * metadata.
 	 *
-	 * @param object $block_metadata The block's metadata containing selectors for
-	 * features.
-	 * @param object $block_node The merged theme.json node for the block.
+	 * @param object $metadata The related block metadata containing selectors.
+	 * @param object $node     A merged theme.json node for block or variation.
 	 * @param object $settings The theme.json settings for the node.
 	 * @param object $theme_json The current theme.json config.
 	 *
-	 * @return array The style declarations for the block's features with custom
+	 * @return array The style declarations for the node's features with custom
 	 * selectors.
 	 */
-	protected static function get_feature_declarations_for_block( $block_metadata, &$block_node, $settings, $theme_json ) {
+	protected static function get_feature_declarations_for_node( $metadata, &$node, $settings, $theme_json ) {
```

Comment:
The `$theme_json` parameter is always the own `$this->theme_json` instance. Can we remove the parameter?

Reply:
Actually, same for the `$settings` parameter: it can be removed and accessed through the `$this->theme_json` within the function itself.

Reply:
I've removed these two params in https://github.com/WordPress/gutenberg/pull/49393/commits/60e6148f00ec1f57434093f1e9ca8100cb675d00. Let me know if there was anything else you had in mind.